### PR TITLE
Add some missing helper classes

### PR DESCRIPTION
**Pull request by tylerbenson**
Missing these helpers made the instrumentation fail.

I also included a bunch more tests covering spring data+elasticsearch.

### CODE REVIEW

File: dd-java-agent/instrumentation/elasticsearch-transport-2/src/main/java/datadog/trace/instrumentation/elasticsearch2/Elasticsearch2TransportClientInstrumentation.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ public AgentBuilder apply(final AgentBuilder agentBuilder) {
             new HelperInjector(
                 "com.google.common.base.Preconditions",
                 "com.google.common.base.Joiner",
+                "com.google.common.base.Joiner$1",
+                "com.google.common.base.Joiner$2",
+                "com.google.common.base.Joiner$MapJoiner",
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/elasticsearch-transport-2/src/main/java/datadog/trace/instrumentation/elasticsearch2/Elasticsearch2TransportClientInstrumentation.java]: These were the rest of the missing helpers.

File: dd-java-agent-ittests/src/test/java/datadog/trace/agent/test/IntegrationTestUtils.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public static ClassLoader getBootstrapProxy() throws Exception {
    * @throws IOException
    */
   public static URL createJarWithClasses(final Class<?>... classes) throws IOException {
-    final File tmpJar = File.createTempFile(UUID.randomUUID().toString() + "", ".jar");
+    final File tmpJar = File.createTempFile(UUID.randomUUID().toString() + "-", ".jar");
```
Comment (realark) [inline dd-java-agent-ittests/src/test/java/datadog/trace/agent/test/IntegrationTestUtils.java]: Just curious, why was this needed?

Comment (tylerbenson) [reply dd-java-agent-ittests/src/test/java/datadog/trace/agent/test/IntegrationTestUtils.java]: It wasn't... It just makes the resulting file name nicer.  Instead of `<uuid><somerandomnumber>.jar` you get `<uuid>-<somerandomnumber>.jar`.  Want me to revert it?
